### PR TITLE
Feature/detail chart

### LINKED ISSUE #9: Storybook: Chart Detail
When I click on a specific sensor under Dashboard button on the sidebar menu like for example PH or EC, I want to be able see chart only for that specific sensor.

Create a **LINE CHART** component like the one below that enables the date range to be changed by moving that bar underneath

Screenshot 2020-03-06 at 15 33 15

Note: Create the components with the appropriate style. Please check the wireframes https://www.figma.com/file/BGeghDJzrIq4rq8QsbGa5p/Seasony-final-flow?node-id=0%3A1

### PR DESCRIPTION
**Pull request by MirSzabo**
resolves #9

### CODE REVIEW

File: src/client/components/page-detail-chart/page-detail-chart.style.css [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.detail-page {
+  background: #e5e5e5;
```
[the remark this answers is not in the document]
Comment (MirSzabo) [reply src/client/components/page-detail-chart/page-detail-chart.style.css]: corrected

File: src/client/components/page-detail-chart/page-detail-chart.style.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  font-style: normal;
+  font-weight: bold;
+  font-size: 32px;
+  line-height: 39px;
+
+  display: flex;
+  align-items: center;
+  text-transform: uppercase;
+
+  color: #3b4349;
```
[the remark this answers is not in the document]
Comment (MirSzabo) [reply src/client/components/page-detail-chart/page-detail-chart.style.css]: corrected

File: src/client/components/page-detail-chart/page-detail-chart.style.css [hunk 3]
```diff
@@ -0,0 +1,16 @@
+.detail-page {
+  background: #e5e5e5;
+}
+.detail-page-header {
+  font-family: Proxima Nova;
+  font-style: normal;
+  font-weight: bold;
+  font-size: 32px;
+  line-height: 39px;
```
[the remark this answers is not in the document]
Comment (MirSzabo) [reply src/client/components/page-detail-chart/page-detail-chart.style.css]: this mean I should delete whole styling for header here and use what will be in theme file?

File: src/client/components/detail-chart/detail-chart.component.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    <div className="App">
+      <ResponsiveContainer
+        className="detail-chart-container"
+        width="100%"
+        height={400}
+      >
+        <LineChart data={lettuce} margin={{ right: 80 }}>
+          <XAxis
+            stroke="#DDE0E3"
+            dataKey="timestamp"
```
[the remark this answers is not in the document]
Comment (MirSzabo) [reply src/client/components/detail-chart/detail-chart.component.js]: I would wait for new json file with data and based on format of dates there I will progress. We agreed on Sunday class that we will not use 1, 2, 3 values so I will postpone this for now.

File: src/client/components/detail-chart/detail-chart.component.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            height={50}
```
[the remark this answers is not in the document]
Comment (MirSzabo) [reply src/client/components/detail-chart/detail-chart.component.js]: whole chart is inside <ResponsiveContainer> so it should stay proportional

File: src/client/components/detail-chart/detail-chart.component.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            dataKey="temp"
+            stroke="#000000"
+            fill="#000000"
+            strokeWidth={2}
+          />
+
+          <Brush
+            className="range-bar"
+            startIndex={0}
+            endIndex={50}
```
[the remark this answers is not in the document]
Comment (MirSzabo) [reply src/client/components/detail-chart/detail-chart.component.js]: yes, I think we will make some calculation for startIndex and endIndex
but data file we used was quite big so it looked better to have endIndex in the middle but maybe with real data it doesn't make sense

File: src/client/components/page-detail-chart/page-detail-chart.style.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (MirSzabo) [reply src/client/components/page-detail-chart/page-detail-chart.style.css]: I commit the code without header styling so I hope it is ok

File: src/client/components/detail-chart/detail-chart.component.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (MirSzabo) [reply src/client/components/detail-chart/detail-chart.component.js]: I have new json file with new format of date so I will continue with formatting X-axis

File: src/client/components/detail-chart/detail-chart.component.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+              fontSize: 10,
+              fill: '#666666',
+            }}
+            fill="none"
+            ticks={[10, 15, 20, 25, 30]}
+          />
+          <Tooltip />
+          <ReferenceLine y={20} stroke="#6F9C67" strokeWidth={2} />
+          <ReferenceLine y={15} stroke="#9C5256" strokeWidth={2} />
+          <ReferenceLine y={25} stroke="#9C5256" strokeWidth={2} />
```
Comment (bytewiz) [inline src/client/components/detail-chart/detail-chart.component.js]: @MirSzabo As we talked about let's make these props and then we are good to go! I will Approve it and you can add this fix 👍

Comment (MirSzabo) [reply src/client/components/detail-chart/detail-chart.component.js]: I added props exactly as we discussed during online class (also in more lines)

**Comment by dpfernandes** [comment]
Approved

**Pull request merged**